File: ZigZag COnversion/ZigZag.java
public class ZigZag {
    public String convert(String s, int numRows){
        //Define StringBuilders
        StringBuilder[] sbs = new StringBuilder[numRows];
        for(int i = 0; i<numRows; i++){
            sbs[i] = new StringBuilder();
        }
        //Define Variables
        char[] arr = s.toCharArray();
        int n = arr.length;
        int index =0;
        //Tranverse zig zig
    while(index <n){
        //GO down
        for(int j=0; j<numRows && index < n; j++){
            sbs[j].append(arr[index++]);
        }
        for (int j=numRows -2; j>0 && index <n; j--){
            sbs[j].append(arr[index++]);
        }

    }
    //combine all stringbuiders into one
    StringBuilder res = sbs[0];
    for(int i = 1; i<numRows; i++){
        res.append(sbs[i].toString());
    }
    return res.toString();
}
}

    
    

    
    

File: ZigZag COnversion/ZigZag.py
class Solution:
    def convert(self, s: str, numRows: int) -> str:
        # Define list of strings
        sbs = ['' for _ in range(numRows)]

        # Define variables
        arr = list(s)
        n = len(arr)
        index = 0

        # Traverse zigzag
        while index < n:
            # Go down
            for j in range(numRows):
                if index < n:
                    sbs[j] += arr[index]
                    index += 1

            # Go up
            for j in range(numRows - 2, 0, -1):
                if index < n:
                    sbs[j] += arr[index]
                    index += 1

        # Combine all strings into one
        res = ''.join(sbs)
        return res
